spyder/plugins/ipythonconsole/plugin.py

```diff
@@ -202,6 +202,9 @@ def apply_plugin_settings(self, options):
                 client.reset_warning = reset_namespace_o
             if ask_before_restart_n in options:
                 client.ask_before_restart = ask_before_restart_o
+            client.shellwidget.set_pdb_ignore_lib()
+            client.shellwidget.set_pdb_execute_events()
+            client.shellwidget.set_pdb_use_exclamation_mark()

```

Comment:
Since you moved the settings for these options to the IPython console, you need to apply them like this now:

```python
...
pdb_ignore_lib_n = 'pdb_ignore_lib'
pdb_ignore_lib_o = self.get_option(pdb_ignore_lib_n)
for client in self.clients:
   ...
   if pdb_ignore_lib_n in options:
      client.shellwidget.set_pdb_ignore_lib(pdb_ignore_lib_o)
```